### ISSUE TITLE
UDP server is not working with me

### ISSUE THREAD

**Issue by hishamBakr** [swiftsocket/SwiftSocket#6]
Thank you very much for great effort.

I connect to Java server. server receive string message correctly.

when server reply to me with string message, callback is called but data,remoteip,remoteport are nil

I use this code : 
func testudpserver(){
        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), { () -> Void in
            var server:UDPServer=UDPServer(addr:"127.0.0.1",port:19876)
            var run:Bool=true
            while run{
                var (data,remoteip,remoteport)=server.recv(1024)
                println("recive")
                if let d=data{
                    if let str=String(bytes: d, encoding: NSUTF8StringEncoding){
                        println(str)
                    }
                }
                println(remoteip)
                server.close()
                break
            }
        })
    }

**Comment by xyyc** [comment]
do you use this library as client or server?
UDPClient(swift)   --> connect-->  java server
or java client  -->connect--> UDPClient(swift)  ?

**Comment by xyyc** [comment]
@hishamBakr 
do you use this library as client or server?
UDPClient(swift) --> connect--> java server
or java client -->connect--> UDPClient(swift) ?

**Comment by hishamBakr** [comment]
Thanks for reply,
 I use library as server 
java client -->connect--> UDPClient(swift)

I receive message but null.

I switched now to another library called "CocoaAsyncSocket" it is objective c but I used in in Swift and now I receive string message correctly

**Comment by xyyc** [comment]
thanks for your feedback, i will check it.

**Comment by xyyc** [comment]
@hishamBakr thanks for your feedback, i will check it.

**Comment by yamiyukiharu** [comment]
Hi, i seemed to found a line of code that prevents UdpServer from working properly, in yudpsocket.c line 60:
          memset( &serv_addr, '\0', sizeof(serv_addr));
here you are wiping out all the data in ser_addr, which makes the socket bind to an address of 0.0.0.0
UdpServer is working fine for me once i removed that line. 
Thanks a lot for this library!

**Comment by iain17** [comment]
Yeah got the same problem here. The tests even written (testudpserver, testudpclient) don't work. @yamiyukiharu that line seems to be gone now, but it still doesn't receive on the other side.

**Comment by segabor** [comment]
I can confirm the fix above works. @iain17 thanks!
